store/migration/10001__init_schema.sql

```diff
@@ -59,6 +59,63 @@ VALUES
         ''
     );
 
+-- auth provider table store the config of 3rd party auth provider
+CREATE TABLE auth_provider (
+    id         INTEGER PRIMARY KEY AUTOINCREMENT,
+    row_status TEXT    NOT NULL CHECK (
+        row_status IN ('NORMAL', 'ARCHIVED')
+        )                       DEFAULT 'NORMAL',
+    creator_id INTEGER NOT NULL REFERENCES principal (id),
+    created_ts BIGINT  NOT NULL DEFAULT (strftime('%s', 'now')),
+    updater_id INTEGER NOT NULL REFERENCES principal (id),
+    updated_ts BIGINT  NOT NULL DEFAULT (strftime('%s', 'now')),
+    `type`     TEXT    NOT NULL CHECK (`type` IN ('GITLAB_SELF_HOST')),
+    -- payload is determined by the type of the auth provider
+    payload    TEXT    NOT NULL
+);
+
+CREATE TRIGGER IF NOT EXISTS `trigger_update_auth_provider_modification_time`
+    AFTER
+        UPDATE
+    ON `auth_provider`
+    FOR EACH ROW
+BEGIN
+    UPDATE
+        `auth_provider`
+    SET updated_ts = (strftime('%s', 'now'))
+    WHERE rowid = old.rowid;
+END;
+
+-- third parties account
+CREATE TABLE third_party_account (
+    id           INTEGER PRIMARY KEY AUTOINCREMENT,
+    principal_id INTEGER NOT NULL REFERENCES principal (id),
+    row_status   TEXT    NOT NULL CHECK (
+        row_status IN ('NORMAL', 'ARCHIVED')
+        )                         DEFAULT 'NORMAL',
+    creator_id   INTEGER NOT NULL REFERENCES principal (id),
+    created_ts   BIGINT  NOT NULL DEFAULT (strftime('%s', 'now')),
+    updater_id   INTEGER NOT NULL REFERENCES principal (id),
+    updated_ts   BIGINT  NOT NULL DEFAULT (strftime('%s', 'now')),
+    `type`       TEXT    NOT NULL CHECK (`type` IN ('GITLAB_SELF_HOST')),
+    -- payload is determined by the type of the third_party_account.
+    payload      TEXT    NOT NULL
+);
+
+CREATE INDEX idx_third_party_account_type ON third_party_account (`type`);
```

Comment:
we don't need to have an index on type since it's all the same value.

On the other hand, we probably need to have an index on principal_id